### PR TITLE
Added conditional check for tabindex in focusRedirector

### LINKED ISSUE #5047: When using Talkback, multiple swipes are required to navigate between messages in the transcript.
### Is it an issue related to Adaptive Cards?

No.

### What is the PWD impact?

Screen reader users will be impacted as the users get confused and need to swipe many times while navigating. The screen reader users efforts will be increased.

### What browsers and screen readers do this issue affect?

Android: Chrome with TalkBack

### Are there any code-based customization done to Web Chat?

No, I am using Web Chat without any customizations except "styleOptions".

### What version of Web Chat are you using?

Latest production

### Which area does this issue affect?

Transcript navigation

### What is the public URL for the website?

_No response_

### How to reproduce the issue?

1. With an Android device, navigate to https://compulim.github.io/webchat-loader/.
2. Select the preset "[Public] MockBot" and start a webchat.
3. Send at least two messages.
4. Enable TalkBack if not done already.
5. Tap the first message to focus it. TalkBack should read out its content.
6. With the first message in focus, swipe right to move focus to the next message.

### What do you expect?

Swiping right once should move focus to the next message.

### What actually happened?

Moving focus to the next message required five swipes.

1. After the first swipe, the timestamp for the first message was read out.
2. After the second swipe, nothing was read out.
3. After the third swipe, nothing was read out.
4. After the fourth swipe, "Bot said colon" was read out.
5. After the fifth swipe, the content of the second message was read out.

### **Environment Details:**

- Application Name: Chat widget
- URL: https://ppe2dot1with5dot1serorg3.crm10.dynamics.com/
- App name: Customer Service admin centerC
- Device: Android Version 14.
- Organization name: ppe2dot1with5dot1serorg3
- Server version: 9.2.23053.00184
- Client version: 1.4.5733-2305.3
- #URL: [React App (windows.net)](https://inapp.blob.core.windows.net/tools/omniwidget/index.html)
- Browser: Chrome

### Do you have any screenshots or recordings to repro the issue?

https://github.com/microsoft/BotFramework-WebChat/assets/120019572/eb5b853e-4a23-4101-a1c6-4b0cb0cf46a7

### Did you find any DOM elements that might have caused the issue?

After the second swipe when nothing is read out, I suspect TalkBack is reading the second FocusRedirector in the FocusTrap wrapping the first message. https://github.com/microsoft/BotFramework-WebChat/blob/58ec6cd3fcd19a8b070f9afad4ae05e6ab6881a4/packages/component/src/Transcript/FocusTrap.tsx#L56C1-L57C1

After the third swipe when nothing is read out, I suspect TalkBack is reading the first FocusRedirector in the FocusTrap wrapping the second message. https://github.com/microsoft/BotFramework-WebChat/blob/58ec6cd3fcd19a8b070f9afad4ae05e6ab6881a4/packages/component/src/Transcript/FocusTrap.tsx#L52C1-L53C1

### MAS reference

MAS 1.3.2 – Meaningful Sequence

### WCAG reference

_No response_

### WAI-ARIA reference

_No response_

### Adaptive Card JSON

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amirmursal**
> Fixes #5047

## Changelog Entry

 Fixes [#5047] Added conditional check for tabindex in focusRedirector [#5223], by @amirmursal

## Description

Talkback is announcing empty div while moving to next message. This fixed after refactoring the FocusRedirector component 

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/Utils/FocusRedirector.tsx [hunk 1]
```diff
@@ -30,8 +35,14 @@ const FocusRedirector: FC<FocusRedirectorProps> = ({ className, onFocus, redirec
...
   //             This redirector is designed to capture TAB only and should not react on browse mode.
   //             However, reacting with browse mode is currently okay. Just better to leave it alone.
-
-  return <div className={className} onFocus={handleFocus} tabIndex={0} />;
+  return (
+    <div
+      aria-hidden={ariaHidden}
+      className={className}
+      onFocus={handleFocus}
+      tabIndex={redirectRef?.current ? TAB_INDEX_ENABLED : TAB_INDEX_DISABLED}
```
Comment (OEvgeny) [inline packages/component/src/Utils/FocusRedirector.tsx:43]: It is ok to use just:
```suggestion
      tabIndex={redirectRef?.current ? 0 : -1}
```

If lint fails, can disable the rule for the file

File: packages/component/src/Utils/FocusRedirector.tsx [hunk 2]
```diff
@@ -30,8 +35,14 @@ const FocusRedirector: FC<FocusRedirectorProps> = ({ className, onFocus, redirec
   //             When using NVDA in browse mode, press up/down arrow keys will focus on this redirector.
   //             This redirector is designed to capture TAB only and should not react on browse mode.
   //             However, reacting with browse mode is currently okay. Just better to leave it alone.
-
-  return <div className={className} onFocus={handleFocus} tabIndex={0} />;
+  return (
+    <div
+      aria-hidden={ariaHidden}
```
Comment (OEvgeny) [inline packages/component/src/Utils/FocusRedirector.tsx:40]:
```suggestion
      aria-hidden={redirectRef?.current ? 'false' : 'true'}
```

File: packages/component/src/Utils/FocusRedirector.tsx [hunk 3]
```diff
@@ -19,6 +19,11 @@ type FocusRedirectorProps = {
 };
 
 const FocusRedirector: FC<FocusRedirectorProps> = ({ className, onFocus, redirectRef }) => {
+  // Hide from screen readers when not needed
+  const ariaHidden = !redirectRef?.current ? 'true' : 'false';
+  const TAB_INDEX_DISABLED = -1;
+  const TAB_INDEX_ENABLED = 0;
+
```
Comment (OEvgeny) [inline packages/component/src/Utils/FocusRedirector.tsx:26]:
```suggestion
  // Hide from screen readers when not in use to prevent unnecessary announcements
  const shouldHide = !!redirectRef?.current;

```

File: __tests__/__image_snapshots__/chrome-docker/adaptive-cards-js-broken-card-of-invalid-version-1-snap.png
Comment (OEvgeny) [inline __tests__/__image_snapshots__/chrome-docker/adaptive-cards-js-broken-card-of-invalid-version-1-snap.png:1]: Unrelated?

**Comment by OEvgeny** [comment]
Please fix tests

File: __tests__/__image_snapshots__/html/transcript-navigation-scroll-into-view-mouse-js-transcript-navigation-when-focusing-on-transcript-using-mouse-should-not-scroll-focused-activity-into-view-2-snap.png
Comment (OEvgeny) [inline __tests__/__image_snapshots__/html/transcript-navigation-scroll-into-view-mouse-js-transcript-navigation-when-focusing-on-transcript-using-mouse-should-not-scroll-focused-activity-into-view-2-snap.png:1]: This looks like a failed case: why focus order did change?

File: packages/component/src/Utils/FocusRedirector.tsx (hunk 1 above)
Comment (amirmursal) [reply packages/component/src/Utils/FocusRedirector.tsx]: fixed

File: packages/component/src/Utils/FocusRedirector.tsx (hunk 3 above)
Comment (amirmursal) [reply packages/component/src/Utils/FocusRedirector.tsx]: fixed

File: packages/component/src/Utils/FocusRedirector.tsx (hunk 2 above)
Comment (amirmursal) [reply packages/component/src/Utils/FocusRedirector.tsx]: fixed

File: __tests__/__image_snapshots__/html/transcript-navigation-scroll-into-view-mouse-js-transcript-navigation-when-focusing-on-transcript-using-mouse-should-not-scroll-focused-activity-into-view-2-snap.png
Comment (amirmursal) [reply __tests__/__image_snapshots__/html/transcript-navigation-scroll-into-view-mouse-js-transcript-navigation-when-focusing-on-transcript-using-mouse-should-not-scroll-focused-activity-into-view-2-snap.png:1]: some of the test case are failing. I am yet to figure out what is causing this issue. Irrelevant snapshots are getting updated. I might need to raise a new pull request
